File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    stack<char>st;
    
    for(int i=0; i<s.length(); i++)
    {
        if(s[i] == '(' )
        {
            st.push(s[i]);
        }
        else if (s[i] == ')')
        {
            if(!st.empty() && st.top() == '(')
            {
                st.pop();   
            }
            else
                return false;
                
        }
    }
    return st.empty();
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> st = new Stack<>();
        
        for(char ch : s.toCharArray()){
            if(ch == '('){
                st.push(ch);
            }else if(ch== ')'){
                if(!st.isEmpty()) st.pop();
                else return false;
            }
        }

        return st.isEmpty();
    }
}